### ISSUE TITLE
Checkpoint architecture and validation protocol do not match the released code/paper results

### ISSUE THREAD

**Issue by abdinlin22** [SEU-zxj/LLM-Augmented-MTR#7]
Hello,

Thank you for uploading the pretrained checkpoints. I am trying to reproduce the 20% WOMD result reported in the paper, but after inspecting the released checkpoint, source code, configuration files, and training log, I found several apparent inconsistencies.

## 1. Checkpoint architecture does not match the current source code

I evaluated the following checkpoint:

```text
checkpoints/20_percent/LLM-Augmented-MTR/checkpoint_epoch_29.pth
```

The checkpoint contains LLM-specific parameters such as:

```text
motion_decoder.intention_hint_decoder.layers.*
motion_decoder.intention_hint_hand_encode_align_layer.weight
motion_decoder.intention_hint_hand_encode_align_layer.bias
```

However, these modules do not exist in the current GitHub source code.

The current implementation instead defines:

```text
motion_decoder.context_proj_layer.0.weight
motion_decoder.context_proj_layer.0.bias
motion_decoder.context_proj_layer.2.weight
motion_decoder.context_proj_layer.2.bias
```

These `context_proj_layer` parameters are not present in the uploaded checkpoint.

Since `load_params_from_file()` filters unmatched parameters and calls:

```python
self.load_state_dict(model_state_disk, strict=False)
```

the evaluation continues without raising an error. However, the old `intention_hint_*` parameters are ignored, while the current `context_proj_layer` remains randomly initialized.

Therefore, although inference completes successfully, it appears that the uploaded LLM checkpoint is not compatible with the currently released model architecture.

## 2. The checkpoint training log refers to an unreleased implementation

The training log associated with the checkpoint refers to the following configuration and options:

```text
mtr+20_percent_data_training_has_GPT_intention.yaml
WITH_INTENTION_HINT: True
INTEENTION_OPTIMIZATION: weight
```

It also refers to files such as:

```text
all_types_map.pkl
train_standardAngle_align_GPT_with_GT.pkl
valid_standardAngle_all_types_map_align_GPT.pkl
handEncodeMatrix.json
```

I could not find these files, the corresponding YAML configuration, or the `WITH_INTENTION_HINT` implementation in the current repository or its public commit history.

The currently released implementation instead uses:

```text
LOAD_CONTEXT_DATA
RETRIEVAL_WINDOW_SIZE
ENCODER_FOR_CONTEXT
context_proj_layer
```

Could you please clarify whether the uploaded checkpoint was trained using an older version of the project?

If so, would it be possible to share the exact source code and configuration corresponding to the checkpoint, particularly:

```text
mtr+20_percent_data_training_has_GPT_intention.yaml
intention_hint_decoder implementation
intention_hint_hand_encode_align_layer implementation
handEncodeMatrix.json
```

## 3. Validation protocol appears to use only 5,864 scenarios

The checkpoint training log reports evaluation on:

```text
5,864 validation scenarios
14,839 evaluated agents
```

However, the complete WOMD validation set processed by the released MTR pipeline contains:

```text
44,097 validation scenarios
192,172 evaluated agents
```

The released 20% configuration uses:

```yaml
SAMPLE_INTERVAL:
    train: 5
    valid: 1
    test: 1
```

Therefore, according to the current configuration, only the training set is subsampled, while evaluation should use the complete validation set.

Could you please clarify how the 5,864 validation scenarios were selected?

In particular, could you share one of the following?

* The exact validation scenario ID list
* The original validation info/mapping file
* The filtering script used to construct this subset
* The missing `valid_standardAngle_all_types_map_align_GPT.pkl` file

It seems possible that evaluation was performed only on scenarios or agents for which LLM annotations were available, but I could not confirm this from the released code.

## 4. Paper metrics and checkpoint-log metrics are not identical

The paper reports the following 20% LLM-Augmented-MTR results:

```text
                 mAP       minADE      minFDE      MissRate
Vehicle          0.3954
Pedestrian       0.3754
Cyclist          0.2924
Average          0.3527     0.6693      1.3580      0.1664
```

However, the evaluation recorded in the uploaded checkpoint log reports:

```text
                 mAP       minADE      minFDE      MissRate
Vehicle          0.4028
Pedestrian       0.3621
Cyclist          0.2930
Average          0.3527     0.6795      1.3695      0.1700
```

Although the average mAP is the same, the class-wise mAP values and the other metrics differ.

Could you please confirm whether the uploaded checkpoint is the model used for the paper’s Table III, or whether it belongs to a different experiment?

## 5. Result obtained with the current released code

For completeness, I evaluated the uploaded 20% LLM checkpoint using:

* Current repository commit: `67229c4373a5cf1cf6c7464dbfd3ad391181d75e`
* Full WOMD validation set: 44,097 scenarios
* `context_data_encoder_20.pkl`
* `RETRIEVAL_WINDOW_SIZE: 4`
* `ENCODER_FOR_CONTEXT: 20`

The resulting metrics were:

```text
                 mAP       minADE      minFDE      MissRate
Vehicle          0.3766     0.8406      1.6672      0.1817
Pedestrian       0.3581     0.3830      0.8100      0.0940
Cyclist          0.2607     0.8939      1.8336      0.2665
Average          0.3318     0.7058      1.4369      0.1807
```

However, I do not consider this a valid reproduction because the current `context_proj_layer` parameters are missing from the checkpoint and remain randomly initialized.

## Questions

Could you please clarify the following?

1. Which exact source-code version was used to train the uploaded checkpoints?
2. Is the uploaded 20% checkpoint the checkpoint used for Table III in the paper?
3. Could you share the older `WITH_INTENTION_HINT` implementation and its YAML configuration?
4. Could you share the exact 5,864-scenario validation subset or the script/file used to construct it?
5. Should the uploaded checkpoint be evaluated with the old intention-hint pipeline rather than the current retrieval-context pipeline?

Providing the exact code/configuration and validation split corresponding to the uploaded checkpoint would make it possible to reproduce the reported result correctly.

Thank you very much for your time and for releasing the project.
